### PR TITLE
Apk upload modifications 

### LINKED ISSUE #1180: APKs in apk branch not updating
**Actual Behaviour**

Ever since PR #1033 merge, the APKs in apk branch are not updating. 
ref All Circle CI merge builds after [this](https://circleci.com/gh/fossasia/susi_android/2262) build are failing.

**Expected Behaviour**

Currently, do not know the reason. But expected behaviour should obviously be that the APKs should update. 

**LogCat for the issue**
```
./exec/apk-gen.sh

apk    build.gradle  exec		gradlew      nohup.out
app    circle.yml    gradle		gradlew.bat  README.md
build  docs	     gradle.properties	LICENSE      settings.gradle
cp: cannot stat ‘/home/ubuntu/susi_android/app/build/outputs/apk/app-debug.apk’: No such file or directory

./exec/apk-gen.sh returned exit code 1
```

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1180 

Changes: added three separate run tasks in circleci config file

### CODE REVIEW

**Comment by batbrain7** [comment]
screen shot 2018-05-22 at 11 23 11 am

@iamareebjamal this is shown in the circle ci config?

**Comment by iamareebjamal** [comment]
This is not related to build scripts. This is circle's artifact upload system

File: circle.yml [hunk 1]
```diff
@@ -26,7 +26,11 @@ jobs:
       - run:
           command:
             bash exec/prep-key.sh
+      - run:
+          command:
             chmod +x exec/apk-gen.sh
+      - run:
+          command:
```
Comment (iamareebjamal) [inline circle.yml]: Merge these two into one like `bash exec/apk-gen.sh`

File: circle.yml [hunk 2]
```diff
@@ -26,6 +26,8 @@ jobs:
       - run:
           command:
             bash exec/prep-key.sh
+      - run:
+          command:
             chmod +x exec/apk-gen.sh
             ./exec/apk-gen.sh
```
Comment (iamareebjamal) [inline circle.yml]: Not this. `bash exec/apk-gen.sh`

Comment (batbrain7) [reply circle.yml]: @iamareebjamal I combined it in the second commit but the CI showed an error of permission denied

File: circle.yml [hunk 3]
```diff
@@ -25,7 +25,9 @@ jobs:
           command: ./gradlew build
       - run:
           command:
-            bash exec/prep-key.sh
+            bash exec/apk-gen.sh
+      - run:
+          command:
             chmod +x exec/apk-gen.sh
             ./exec/apk-gen.sh
```
Comment (iamareebjamal) [inline circle.yml]: Still same as previous commit

Should be `bash exec/apk-gen.sh`

File: circle.yml [hunk 4]
```diff
@@ -25,7 +25,9 @@ jobs:
           command: ./gradlew build
       - run:
           command:
-            bash exec/prep-key.sh
+            bash exec/apk-gen.sh
```
Comment (iamareebjamal) [inline circle.yml]: This should be `bash exec/prep-key.sh`

File: circle.yml (hunk 3 above)
Comment (batbrain7) [reply circle.yml]: removing statements 
chmod +x exec/apk-gen.sh
./exec/apk-gen.sh

and adding bash exec/apk-gen.sh ?

Comment (iamareebjamal) [reply circle.yml]: Yes

**Comment by batbrain7** [comment]
@iamareebjamal can you please tell what is the difference b/w the statements ?

**Comment by iamareebjamal** [comment]
bash is interpreter just like python. If you want to run python script, you write `python script.py` it runs even if there is no executable permission on the file because you are explicitly invoking python which itself is executable

Whereas, you can also run the python script as `./script.py` only you need shebang telling `#!/bin/python` on top so that the shell knows which interpreter to send the script to. Also, you need to have executable permission on the file. This is very restricted with shebang and permission, that's why it's just better to use `bash script.sh` or `python script.py`

**Comment by iamareebjamal** [comment]
One more change is required. The apk-gen script does not have the correct version of build tools in it. Please fix it

**Comment by batbrain7** [comment]
@iamareebjamal where to find the apk-gen script?

**Comment by iamareebjamal** [comment]
You have written its path in the run command in build script

**Pull request merged**